process/smartContract/processorV2/failureContext.go

```diff
@@ -0,0 +1,58 @@
+package processorV2
+
+import vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+
+type failureContext struct {
+	sc            *scProcessor
+	processFail   bool
+	txHash        []byte
+	errorMessage  string
+	returnMessage []byte
+	snapshot      int
+	gasLocked     uint64
+	logs          []*vmcommon.LogEntry
+}
+
+func NewFailureContext(sc *scProcessor) *failureContext {
+	return &failureContext{
+		sc:          sc,
+		processFail: false,
+	}
+}
+
+func (failCtx *failureContext) makeSnaphot() *failureContext {
+	failCtx.snapshot = failCtx.sc.accounts.JournalLen()
+	return failCtx
+}
+
+func (failCtx *failureContext) setTxHash(txHash []byte) *failureContext {
```

Comment:
daisy chaining that does not quite add clarity but makes the calls a one-liner.
I always prefer multiple calls on multiple lines as long as they are clear enough.

Reply:
done